App crashes when recording permissions are denied in ActiveSession

Reproduce:

1. Start new Session
2. Click button to start recording
3. Deny permission (recording is not started)
4. go back to main screen
5. App crashes

Stack trace:

```
FATAL EXCEPTION: main
Process: app.musikus, PID: 8757
java.lang.RuntimeException: Unable to stop service app.musikus.services.RecorderService@74b0edd: app.musikus.utils.IllegalRecorderStateException: Can only save recording from paused state
	at android.app.ActivityThread.handleStopService(ActivityThread.java:4871)
	at android.app.ActivityThread.-$$Nest$mhandleStopService(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2297)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:205)
	at android.os.Looper.loop(Looper.java:294)
	at android.app.ActivityThread.main(ActivityThread.java:8177)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
Caused by: app.musikus.utils.IllegalRecorderStateException: Can only save recording from paused state
	at app.musikus.utils.Recorder.delete(Recorder.kt:112)
	at app.musikus.services.RecorderService.onDestroy(RecorderService.kt:276)
	at android.app.ActivityThread.handleStopService(ActivityThread.java:4851)
	at android.app.ActivityThread.-$$Nest$mhandleStopService(Unknown Source:0) 
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2297) 
	at android.os.Handler.dispatchMessage(Handler.java:106) 
	at android.os.Looper.loopOnce(Looper.java:205) 
	at android.os.Looper.loop(Looper.java:294) 
	at android.app.ActivityThread.main(ActivityThread.java:8177) 
	at java.lang.reflect.Method.invoke(Native Method) 
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552) 
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971) 
```

--- a/app/src/main/java/app/musikus/services/RecorderService.kt
+++ b/app/src/main/java/app/musikus/services/RecorderService.kt
@@ -272,11 +272,6 @@ class RecorderService : Service() {
         }
     }
 
-    override fun onDestroy() {
-        recorder.delete()
-        super.onDestroy()
-    }
-
     private fun updateNotification(duration: Duration) {
         val notification: Notification = getNotification(duration)
         val mNotificationManager = getSystemService(NOTIFICATION_SERVICE) as NotificationManager

